cmd/tailscale/cli/up.go

```diff
@@ -76,6 +77,38 @@ var upArgs struct {
 	snat            bool
 	netfilterMode   string
 	authKey         string
+	hostname        string
+}
+
+// validateHostname checks that name is a valid domain name label
+// pursuant to https://tools.ietf.org/html/rfc1034#section-3.1.
+func validateHostname(name string) error {
+	switch {
+	// Empty string is treated as missing hostname and replaced downstream.
+	case len(name) == 0:
+		return nil
+	case len(name) > 63:
+		return fmt.Errorf("longer than 63 characters")
+	}
+
+	name = strings.ToLower(name)
```

Comment:
This means validateHostname accepts the string `"İ"` or `"K"` (`"\xc4\xb0"` or `"\xe2\x84\xaa"`) which aren't ASCII.

Reply:
Here's an example: https://play.golang.org/p/85rNKF57KeH

Reply:
In Turkish letter I news, https://languagelog.ldc.upenn.edu/nll/?p=73 (totally unrelated, kinda)